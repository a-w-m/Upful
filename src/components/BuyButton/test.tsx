import React from 'react'
import {render} from '@testing-library/react'
import BuyButton from "."

describe("BuyButton Component", ()=>{
    test("matches snapshot", ()=>{
        const {container} = render( <BuyButton
            className ='snipcart-add-item'
            data-item-id= ""
            data-item-price= ""
            data-item-name= ""
            data-item-description=""
            data-item-image=""
            data-item-url=""
            data-item-custom1-name=""
            data-item-custom1-options=""
            data-item-custom1-value="">
       </BuyButton>)
        expect(container.firstChild).toMatchSnapshot()

    })
})