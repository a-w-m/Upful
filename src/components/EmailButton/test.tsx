import React from "react"
import { render } from "@testing-library/react"
import EmailButton from "."

const props ={
    title: "test-email"
}

describe("EmailButton component", () => {
  test("matches snapshot", () => {
    const { container } = render(<EmailButton {...props}></EmailButton>)
    expect(container.firstChild).toMatchSnapshot()
  })


  test("component displays correct text", ()=>{
    const { queryByText } = render(<EmailButton {...props}></EmailButton>)
    expect(queryByText("Email us about this product")).toBeTruthy();

  })

  test("component has correct href", ()=>{
    const { getByRole } = render(<EmailButton {...props}></EmailButton>)

    // Click button
    expect(getByRole('link')).toHaveAttribute("href", `mailto:howdy@upful.com?Subject=${props.title}`)


  })

 
})
