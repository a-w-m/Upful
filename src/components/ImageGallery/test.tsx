import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import ImageGallery from "./index"
import { imageGalleryData } from "../../../__mocks__/mock-data"

describe("ImageGallery Component", () => {
  const mockDispatch = jest.fn()

  test("matches snapshot", () => {
    const { container } = render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("dispatch is called when gallery image is clicked ", () => {
    render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    const image = screen.getAllByRole("listitem")[0]
    fireEvent.click(image)
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

test("no list rendered when no images supplied", () => {
  const { container } = render(
    <ImageGallery images={[]} dispatch={jest.fn()}></ImageGallery>
  )
  expect(container.firstChild).toBeNull()
})
