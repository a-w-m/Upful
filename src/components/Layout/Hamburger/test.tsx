import React from "react"
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Hamburger from "."

describe("Hamburger component", () => {
  const setOpen = jest.fn()

  it("should match snapshot", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const button = screen.getByRole("button")
    expect(button).toMatchSnapshot()
  })

  it("should handle click", async () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const user = userEvent.setup()
    const button = screen.getByRole("button")
    await user.click(button)
    expect(setOpen).toHaveBeenCalledTimes(1)
  })

  it("should apply style when open", () => {
    render(<Hamburger open={true} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(45deg)" })
    expect(menu[1]).toHaveStyle({ opacity: "0", transform: "translateX(2rem)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(-45deg)" })
  })

  it("should apply style when not open", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(0)" })
    expect(menu[1]).toHaveStyle({ opacity: "1", transform: "translateX(0)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(0)" })
  })
})
