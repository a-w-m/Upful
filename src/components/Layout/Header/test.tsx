import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import Header from ".."

describe("Header component", () => {
  test("matches snapshot", () => {
    const { container } = render(<Header children="" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders Header with correct site title", () => {
    render(<Header children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
  })

  // test("displays toggle menu on small screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.getByAltText("toggle-menu")).toBeInTheDocument()

  // })

  // test("does not display toggle menu on large screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.queryAllByAltText("toggle-menu")).toBeNull()

  // })
})

// describe("click outside toggle menu", () => {
//   test("Nav remains visible when clicking inside component ", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.click(Nav)
//     expect(Nav).toBeVisible()
//   })

//   test("Nav is not visible when clicking outside component", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.mouseDown(Button)
//     expect(Nav).not.toBeVisible()
//   })
// })
