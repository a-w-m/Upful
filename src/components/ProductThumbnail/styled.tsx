import styled from "styled-components"
import device from "../Layout/mediaQuery"
interface Props {
  stock: number
}

export const Article = styled.article`
  grid-column: span 2;
  /* @media ${device.tablet}{
    width: 400px;
  } */

  /* handle widow/orphan grid items: https://css-irl.info/controlling-leftover-grid-items/ */
  
  /* Dealing with single possible orphan on mobileS, M, L, 4 column grid */
  &:last-child:nth-child(2n-1){
    grid-column-end: -2;
    @media ${device.tablet}{
      grid-column-end: auto;
    }
  }


 /* Dealing with orphan grid items for tablet, laptop, and laptopL, 6 column grid */
 /* Dealing with 2 orphan items */

  &:last-child:nth-child(3n - 1) {
    @media ${device.tablet} {
      grid-column-end: -2;
    }
  }
  &:nth-last-child(2):nth-child(3n + 1) {
    @media ${device.tablet} {
      grid-column-end: 4;
    }
  }
  /* Dealing with single orphan */

  &:last-child:nth-child(3n - 2) {
    @media ${device.tablet} {
      grid-column-end: 5;
    }
  }
`
export const H3 = styled.h3`
  font-size: 1.5rem;
  line-height: 1.25;
  font-family: var(--main-font);
`
export const PriceWrapper = styled.div<Props>`
  font-size: 1.6rem;
  color: #000;
  opacity: 0.5;
  text-decoration: ${props => (props.stock > 0 ? "none" : "line-through")};
`

export const SoldOut = styled.span``

export const ProductInfoContainer = styled.div`
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  align-items: center;
  text-align: center;
  font-size: 1.4rem;
`
