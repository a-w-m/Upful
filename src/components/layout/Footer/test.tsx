import React from 'react'
import {render, screen} from '@testing-library/react'
import Footer from "."


describe("Footer Component", ()=>{

    test("matches component", ()=>{
        const {container} = render(<Footer />)
        expect(container.firstElementChild).toMatchSnapshot()

    })

})