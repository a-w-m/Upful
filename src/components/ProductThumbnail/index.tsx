import React, {useContext} from "react"
import {Context} from "../Provider"
import { graphql, Link } from "gatsby"
import { GatsbyImage } from "gatsby-plugin-image"
import { P } from "../interfaces"
import { Article, H3, PriceWrapper, ProductInfoContainer, SoldOut } from "./styled"



const ProductThumbnail: React.FC<P.Thumbnail> = props => {
  const { id, title, price, image, slug } = props
  const {inventory, isLoading} = useContext(Context)
  return (
    <Article>
      <Link to={slug}>
        <GatsbyImage image={image.childImageSharp.gatsbyImageData} alt="" />
      </Link>
      {!isLoading && <ProductInfoContainer>
        <H3>{title}</H3>
        <PriceWrapper>{price.toFixed(2)}</PriceWrapper>
        {(inventory[id]?.stock <=0) &&<SoldOut>Sold Out </SoldOut> }
      </ProductInfoContainer>}
    </Article>
  )
}

export default ProductThumbnail

export const thumbnailQuery = graphql`
  fragment Thumbnail on MarkdownRemark {
    frontmatter {
      id
      title
      price
      image {
        childImageSharp {
          ...ThumbnailImage
        }
      }
    }
  }
`

export const imageQuery = graphql`
  fragment ThumbnailImage on ImageSharp {
    gatsbyImageData(
      width: 1024
      aspectRatio: 1
      transformOptions: { fit: COVER }
    )
  }
`
