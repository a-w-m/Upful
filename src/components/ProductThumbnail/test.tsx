import React from "react"
import { render, screen } from "@testing-library/react"
import ProductThumbnail from "./index"
import { imageGalleryData } from "../../utils/test/data"

const price = 10
const title = "Pietsie Atlin Blouse"

describe("Thumbnail Component", () => {
  //displays name
  //displays price

  test("price displays correctly", () => {
    render(
      <ProductThumbnail
        price={price}
        title={"Pietsie Atlin Blouse"}
        image={imageGalleryData[0].node}
        slug={"/"}
      ></ProductThumbnail>
    )
    expect(screen.getByText("$10.00")).toBeInTheDocument()
  })

  test("name displays correctly", () => {
    render(
      <ProductThumbnail
        price={0}
        title={"Pietsie Atlin Blouse"}
        image={imageGalleryData[0].node}
        slug={"/"}
      ></ProductThumbnail>
    )
    expect(screen.getByText(title)).toBeInTheDocument()
  })
})
