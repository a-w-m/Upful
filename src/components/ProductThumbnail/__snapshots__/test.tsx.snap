// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail Component matches snapshot 1`] = `
<article
  class="sc-bczRLJ bIIyW"
>
  <a
    href="/haori-coat-in-washed-denim/"
  >
    <div
      class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
      data-gatsby-image-wrapper=""
      style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; height: 280px;"
    >
      <div
        style="max-width: 995px; display: block;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,%3Csvg height='995' width='995' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
          style="max-width: 100%; display: block; position: static;"
        />
      </div>
      <div
        aria-hidden="true"
        data-placeholder-image=""
        style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      />
      <img
        alt="thumbnail"
        data-gatsby-image-ssr=""
        data-main-image=""
        data-src="/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp"
        data-srcset="/static/5e566884fdc0f35caf38e43e3d3b3faa/dc55b/image-1.webp 249w,/static/5e566884fdc0f35caf38e43e3d3b3faa/525ed/image-1.webp 498w,/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp 995w"
        decoding="async"
        loading="lazy"
        sizes="(min-width: 995px) 995px, 100vw"
        style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; object-fit: contain; opacity: 0;"
      />
      <noscript />
      <script
        type="module"
      >
        const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
      </script>
    </div>
  </a>
  <div
    class="sc-eCYdqJ dPRCrd"
  >
    <h3
      class="sc-gsnTZi fcNkwz"
    >
      Haori Coat in Washed Denim
    </h3>
    <div
      class="sc-dkzDqf jPfPxC"
    >
      420.00
    </div>
  </div>
</article>
`;
